_posts/2020-10-04-quarkus-native-on-aarch64.adoc

```diff
@@ -0,0 +1,408 @@
+---
+layout: post
+title: 'Quarkus native running on aarch64'
+date: 2020-10-04
+tags: aarch64 native
+synopsis: Running a Quarkus native app inside a container on aarch64
+author: abattagl
+---
+
+## Intro
+
+When I've been introduced to Quarkus, back in December 2018, I suddenly had the feeling the Quarkus runtime has the potential of covering ~95% of the software use-cases.
+My friend Sanne demoed Quarkus on his laptop when the runtime was still in an alpha state, far from being the large, extended and mature framework everybody knows nowadays. And still, I could experience a new way of think, implement and deploy java apps.
+
+I’ve been recently working on a Hackfest for Red Hat EMEA partners to spread awareness about the potential and the performance of the Quarkus framework. My main technical goal was to demonstrate Quarkus can run everywhere and thanks to its performance capabilities coming from the native compilation process it definitely is the runtime of choice for IoT devices (a.k.a. Far Edge).
+
+To achieve the above goal, I’ve picked up a single-board computer with an integrated ARM-compatible central processing unit and made a Quarkus native app running into a container on top of a Linux based operating system. Here is how:
+
+## Prerequisites
+
+* *Single-board computer:* The ARM-compatible CPU integrated into the single-board computer must support 64-bit mode. 
+
+* *Operating system:* 
+** To make a Quarkus native container image run in a container, a 64-bit Linux OS is required;
+
+** The OS must support aarch64 CPUs, which leads us to the need for an aarch64-base distribution available for that OS;
+
+** Nice to have: the target OS should fully support the target single-board computer.**
+
+* *Container technology:* surprise, surprise! an OCI standard container engine is compulsory to run containers on Linux :-)
+
+* *Build tools:* as per the large amount of resources required by the GraalVM to compile Quarkus in native mode, an additional we need an ARM server / Virtual machine to achieve that goal. The server should meet the following minimum requirements: 
+** CPU: same ARM model as the one embedded in the target single-board computer
+** Cores: 4
+** Ram: 8Gb
+** Disk space: 20Gb
```

Comment:
20gb...damn ;) what is that space used for ?

Reply:
Hypothetically, you'd reuse that VM to work on several projects and build. I'll reduce the min requirement for the disk space and add some considerations about reusing it.

Reply:
Cool